packages/bsky/src/notifications.ts

```diff
@@ -0,0 +1,285 @@
+              recordCid: notif.recordCid,
+            },
+          })
+        } else if (platform === 'android') {
+          notifsToSend.push({
+            tokens: [token],
+            platform: 2,
+            title: title,
+            message: body,
+            topic: appId,
+            data: {
+              reason: notif.reason,
+              recordUri: notif.recordUri,
+              recordCid: notif.recordCid,
+            },
+          })
+        } else {
+          // TODO: Handle web notifs
+        }
+      }
+    }
+    return notifsToSend
+  }
+
+  /**  1. Get the user's token (APNS or FCM for iOS and Android respectively) from the database
+    User token will be in the format:
+        did || token || platform (1 = iOS, 2 = Android, 3 = Web)
+    2. Send notification to `gorush` server with token
+    Notification will be in the format:
+    "notifications": [
+      {
+        "tokens": string[],
+        "platform": 1 | 2,
+        "message": string,
+        "title": string,
+        "priority": "normal" | "high",
+        "image": string, (Android only)
+        "expiration": number, (iOS only)
+        "badge": number, (iOS only)
+      }
+    ]
+    3. `gorush` will send notification to APNS or FCM
+    4.  store response from `gorush` which contains the ID of the notification
+    5. If notification needs to be updated or deleted, find the ID of the notification from the database and send a new notification to `gorush` with the ID (repeat step 2)
+  */
+  async sendPushNotifications(
+    notifications: PushNotification[],
+    pushEndpoint = PUSH_NOTIF_SERVER_URL,
+  ) {
+    // if no notifications, skip and return early
+    if (!notifications || notifications.length === 0) {
+      return
+    }
+    try {
+      await axios.post(
+        pushEndpoint,
+        {
+          notifications: notifications,
+        },
+        {
+          headers: {
+            'Content-Type': 'application/json',
+            accept: 'application/json',
+          },
+        },
+      )
+    } catch (error) {
+      throw new Error('Failed to send push notification')
+    }
+  }
+
+  async registerDeviceForPushNotifications(
+    did: string,
+    platform: Platform,
+    token: string,
+    endpoint = PUSH_NOTIF_SERVER_URL,
+    appId = BSKY_APP_ID,
+  ) {
+    try {
+      // check if token did pair already exists
+      const existing = await this.db.db
+        .selectFrom('notification_push_token')
+        .where('did', '=', did)
+        .where('token', '=', token)
+        .selectAll()
+        .executeTakeFirst()
+      if (existing) {
+        return
+      }
+
+      // if token doesn't exist, insert it
+      await this.db.db
+        .insertInto('notification_push_token')
+        .values({
+          did,
+          token,
+          platform,
+          endpoint,
+          appId,
+        })
+        .execute()
+    } catch (error) {
+      throw new Error('Failed to insert notification token')
```

Comment:
here's another spot where we should avoid eating the error & just let kysely throw so that we can see it in logs

Reply:
Done!